### DISCUSSION TITLE
AdMob onResume/onPause (background)

### DISCUSSION THREAD

**Discussion by Rebsos** [invertase/react-native-firebase#4877] [Q&A]
AdMob's [minimal banner example](https://github.com/googleads/googleads-mobile-android-examples/blob/master/java/admob/BannerExample/app/src/main/java/com/google/android/gms/example/bannerexample/MyActivity.java) calls these functions: **adView.pause()** and **adView.resume()** if the app is brought to the background and back. I can't find this in react-native-firebase/admob.

Does the Banner request more ads in the background (policy?) or drain the battery?
Do we have to unmount the Banner manually onPause and mount it again onResume? (policy?)

**Comment by mikehardy** [comment]
@Rebsos I'm not sure, what did you find when you tested it? What does the code in the module do? https://github.com/invertase/react-native-firebase/tree/master/packages/admob/android

Please report your findings, it could help others

**Comment by Rebsos** [comment]
I did some memory, network and CPU tests with the Android Profiler and meminfo
 
	1.	Good: A mounted banner does not request ads in the background
	2.	Info: A mounted banner, consumes 40 mb memory
	3.	Bad: After unmounting a banner, the 40 mb do not get released
	4.	Good: After unmounting a banner, there are no ad requests anymore
	5.	Info: After mounting a banner and going to the background, the 40 mb do not get released or reduced.
	6.	Good: Unmounting a banner or going with a mounted banner to the background, seems to stop the CPU usage of the banner.
 
 
conclusion:
 
	1.	It should not violate AdMobs policy
	2.	Unmounting a banner won't help reduce memory
 
Bonus:
 
I've put in this in ReactNativeFirebaseAdMobBannerAdViewManager.java for a very quick test, but it seems to have no advantage over "doing nothing"



>             LifecycleEventListener listener = new LifecycleEventListener() {
>             @Override
>             public void onHostResume() {
>                 Log.i(TAG, "- onResume");
>                 if (adView != null) {
>                     adView.resume();
>                 }
>             }
>  
>             @Override
>             public void onHostPause() {
>                 Log.i(TAG, "- onPause");
>                 if (adView != null) {
>                   adView.pause();
>                 }
>             }
>  
>             @Override
>             public void onHostDestroy() {
>               Log.i(TAG, "- onDestroy");
>               if (adView != null) {
>                   adView.destroy();
>               }
>             }
>         };
>         themedReactContext.addLifecycleEventListener(listener);

 
Future:
There may be still an issue with freezing banners. I've seen this in "sbugert/react-native-admob" which I used a very long time (now I switch to this module). Banners often won't refresh, when going back to foreground. Both modules miss "adView.resume()", maybe this is the reason, but I don't know.

**Reply by mikehardy** [reply]
Very interesting - unsure why the pause/resume would be missed if the lifecycle listener was added, lifecycle listeners have one job and they should meet their API contract. The fail to free memory could be a big deal especially if it is somehow additive. Odd that it isn't freed
